pandas-stubs/core/frame.pyi

```diff
@@ -516,7 +516,7 @@ class DataFrame(NDFrame, OpsMixin):
     def T(self) -> DataFrame: ...
     def __getattr__(self, name: str) -> Series: ...
     @overload
-    def __getitem__(self, idx: Scalar | tuple[Hashable, ...]) -> Series: ...
+    def __getitem__(self, idx: Scalar | Hashable | tuple[Hashable, ...]) -> Series: ...
```

Comment:
Could reduce it to `def __getitem__(self, idx: Scalar | Hashable) -> Series: ...` as a `tuple` is declared as hashable in typeshed.

(Many members of `Scalar` are probably also hashable - not sure whether all of them are hashable.)

Reply:
Thanks I just removed `tuple[Hashable, ...]` from the signature.

Reply:
Similarly, should we include a `list[Hashable]` in the 3rd overload?

The 3rd overload:
```python

    @overload
    def __getitem__(
        self,
        idx: Series[_bool]
        | DataFrame
        | Index
        | np_ndarray_str
        | np_ndarray_bool
        | list[_ScalarOrTupleT],
    ) -> DataFrame: ...
```

Reply:
This is actually more complicated than I thought.

> Do we have tests for __getitem__ and slice? 

Yes, there is a slice test, but it is without assert types. 
 https://github.com/pandas-dev/pandas-stubs/blob/fbc5f20d35c0d031fc30dfe2cd8e1dfb21c8610a/tests/test_frame.py#L156

Reply:
I just added more assert_types in the test to make sure the other overloads are still working fine.